src/form/dragDropArea/index.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react'
+import { makeStyles } from '@material-ui/core'
+
+const useStyles = makeStyles({
+  root: {
+    height: '100vh',
+    width: '100%',
+  },
+})
+
+interface IDragDropAreaProps {
+  overlay: React.ReactElement
+  children: React.ReactElement
+  onFilesDrop: (files: File[]) => void
+}
+
+const DragDropArea = ({ overlay, children, onFilesDrop }: IDragDropAreaProps) => {
+  const classes = useStyles()
+  const [dragActive, setDragActive] = React.useState(0)
```

Comment:
shouldn't that be boolean?

Reply:
the overlay doesn't behave properly when this is boolean

Reply:
can we have a type for things like this? if it will be only 0 | 1 can we have something like `type TDragStatus = 0 | 1`so it doesn't look like javascript?

Reply:
or constants maybe?

Reply:
Actually the `dragActive` jumps between values `1` and `2` when the overlay is active and the user drags the file over the area. This is why boolean wouldn't work. The overlay would flicker.